src/registry.py

```diff
@@ -124,7 +124,7 @@ def download(self, namespace: str, name: str, provider: str, version: str) -> st
 
         response = self.s3.list_objects_v2(
             Bucket=self._bucket_name,
-            Prefix=f'{namespace}/{name}/{provider}/{version}'
+            Prefix=f'{namespace}/{name}/{provider}/{version}/'
```

Comment:
I'm not sure this will work correctly - looking at the upload method the objects seem to be stored at the path `{namespace}/{name}/{provider}/{version}.tar.gz`?

Reply:
You're right it is.